wire-ios/Wire-iOS/Sources/UserInterface/Calling/CallDegradationViewController/UIAlertController+DegradedCall.swift

```diff
@@ -23,26 +23,29 @@ import WireSyncEngine
 extension UIAlertController {
 
     static func degradedCall(degradedUser: UserType?, callEnded: Bool = false, confirmationBlock: ((_ continueDegradedCall: Bool) -> Void)? = nil) -> UIAlertController {
+        let title = callEnded ?
+        L10n.Localizable.Call.Degraded.Ended.Alert.title :
+        L10n.Localizable.Call.Degraded.Alert.title
 
-        // Choose localization prefix
-        let prefix = callEnded
-            ? "call.degraded.ended.alert"
-            : "call.degraded.alert"
-
-        // Set message
-        var message = "\(prefix).message"
-
-        switch degradedUser {
-        case .some(let user) where user.isSelfUser:
-            message = "\(message).self".localized
-        case .some(let user):
-            message = "\(message).user".localized(args: user.name ?? "")
-        default:
-            message = "\(message).unknown".localized
+        var message: String
+        if let user = degradedUser {
+            if callEnded {
+                message = user.isSelfUser ?
+                L10n.Localizable.Call.Degraded.Ended.Alert.Message.`self` :
+                L10n.Localizable.Call.Degraded.Ended.Alert.Message.user(user.name ?? "")
+            } else {
+                message = user.isSelfUser ?
+                L10n.Localizable.Call.Degraded.Alert.Message.`self` :
+                L10n.Localizable.Call.Degraded.Alert.Message.user(user.name ?? "")
+            }
+        } else {
+            message = callEnded ?
+            L10n.Localizable.Call.Degraded.Ended.Alert.Message.unknown :
+            L10n.Localizable.Call.Degraded.Alert.Message.unknown
```

Comment:
Can this be easily extracted to a `func`?

Reply:
Maybe then `title` too, to keep the same level of abstraction in the closure

Reply:
Did the same for title and message, thank you!